### ISSUE TITLE
How to used cntk create Grad-cam code?

### ISSUE THREAD

**Issue by kerryJo** [Microsoft/CNTK#3613]
Hi.
I'm Trying Grad-CAM use CNTK 
What should I do That

Also Look at my code Is that right?
Thanks.

I made it by referring to the code below.
https://github.com/jacobgil/keras-grad-cam/blob/master/grad-cam.py

This my code
<code>
def visualize_class_activation_map(conv_outputs,class_weights,img_path, output_path):
        
        original_img = cv2.imread(img_path, 1)
        width, height, _ = original_img.shape        
        #Reshape to the network input shape (3, w, h).
        img = np.array([np.transpose(np.float32(original_img), (2, 0, 1))])       
        conv_outputs = conv_outputs[0, :, :, :]
        #Create the class activation map.       
        weights = np.mean(class_weights, axis = (0, 1))
        cam = np.zeros(dtype = np.float32, shape = conv_outputs.shape[1:3])

        for i, w in enumerate(weights):
                cam += w * conv_outputs[i, :, :]

        cam /= np.max(cam)
        cam = cv2.resize(cam, (height, width))
        heatmap = cv2.applyColorMap(np.uint8(255*cam), cv2.COLORMAP_JET)
        heatmap[np.where(cam < 0.2)] = 0
        img = heatmap*0.5 + original_img
        cv2.imwrite(output_path, img)

# Evaluates a single image using the provided model
def eval_single_image(image_path, image_width, image_height,model):

    ## Alternatively: if you want to use opencv-python
    cv_img = cv2.imread(image_path)
    resized = cv2.resize(cv_img, (image_width, image_height), interpolation=cv2.INTER_NEAREST)
    bgr_image = np.asarray(resized, dtype=np.float32)
    hwc_format = np.ascontiguousarray(np.rollaxis(bgr_image, 2))

    #compute model output
    arguments = {model.arguments[0]: [hwc_format]}
    output = model.eval(arguments)

    output_dict = {}
    for key in output.keys(): output_dict[key.name] = output[key]
    encoder_input  = output_dict[input_node_name]
    encoder_output = output_dict[decoder_node_name]
    decoder_output = output_dict[output_node_name]

    visualize_class_activation_map(encoder_output,decoder_output,image_path,"D://IMAGE//0001.jpg")

def generate_visualization(use_brain_script_model, testing=False):
    
    num_objects_to_eval = 1

    model_file_name = "ResNet18_ImageNet_CNTK.model"
    encoder_output_file_name = "encoder_output_PY.txt"
    decoder_output_file_name = "decoder_output_PY.txt"
   

    # define location of output, model and data and check existence
    output_path = os.path.join(abs_path, "Output")
    model_file = os.path.join("D:\\CNTK\\Examples\\Image\\TransferLearning\\Output\\TransferLearning.model")
    #model_file = os.path.join("D:\\CNTK\\PretrainedModels\\ResNet18_ImageNet_CNTK.model")

    #print_all_node_names(model_file)

    minibatch_source = create_mb_source(test_map_file, image_width, image_height, num_channels, num_classes)

    
    # use this to print all node names in the model
    print_all_node_names(model_file, use_brain_script_model)   

    # load model and pick desired nodes as output
    loaded_model = load_model(model_file)

    output_nodes = combine([loaded_model.find_by_name(input_node_name).owner,
                            loaded_model.find_by_name(decoder_node_name).owner,
                            loaded_model.find_by_name(output_node_name).owner,]) 

    eval_single_image("D:\\IMAGE\\0000.jpg",224,224,output_nodes)

    print("Done. Wrote output to %s" % output_path)

</code>
